passwerk_TMSP/passwerk_TMSP.go

```diff
@@ -171,13 +164,18 @@ func (app *PasswerkApplication) CheckTx(tx []byte) types.Result {
 
 	switch operationalOption {
 	case "writing":
-		if len(parts) < 6 {
+		if len(parts) < 7 {
 			return badReturn("Invalid number of TX parts")
 		}
 	case "deleting":
-		if len(parts) < 3 {
+		if len(parts) < 7 { //note that the length for deleting and writing just happens to be the same, may change in future passwerk
 			return badReturn("Invalid number of TX parts")
 		}
+
+		//check to make sure that system state hasn't changed
+		if parts[3] != bytes2HexString(app.state.Hash()) { //here parts[3] has passed on the system state from the broadcast

```

Comment:
I don't understand this. parts[3] is the hashed password I thought? besides, ensuring the state root is exactly what you thought is gonna result in a lot of invalid txs, tho it could be a nice feature.